test/abstractarray.jl

```diff
@@ -1315,6 +1315,45 @@ Base.pushfirst!(tpa::TestPushArray{T}, a::T) where T = pushfirst!(tpa.data, a)
     @test tpa.data == reverse(collect(1:6))
 end
 
+mutable struct SimpleArray{T} <: AbstractVector{T}
+    els::Vector{T}
+end
+Base.size(sa::SimpleArray) = size(sa.els)
+Base.getindex(sa::SimpleArray, idx...) = getindex(sa.els, idx...)
+Base.setindex!(sa::SimpleArray, v, idx...) = setindex!(sa.els, v, idx...)
+Base.resize!(sa::SimpleArray, n) = resize!(sa.els, n)
+
+@testset "Failing `push!` leaves size unchanged" begin
+    a = SimpleArray{String}(String[])
+    # single-arg case
+    try
+        push!(a, 1)
+    catch
+        @test isempty(a)
```

Comment:
These need to be outside, or in `finally`, otherwise this test will pass if the `try` block doesn't throw.

Reply:
Good point, thanks!